File: AndroidProtobufTest/app/src/main/java/com/pix/protobufandroid/MainActivity.java
package com.pix.protobufandroid;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.pix.protobufandroid.protobuf.Login;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //构建对象
        Login login = new Login.Builder()
                .cmd("login")
                .name("张三")
                .passwd("123456")
                .userid(7369L).build();

        Log.d(TAG,"encode," + login.toString());

        //序列化
        byte [] data = login.encode();


        //反序列化
        try {
            Login deLogin = Login.ADAPTER.decode(data);
            Log.d(TAG,"deLogin,cmd:" + deLogin.cmd);
            Log.d(TAG,"deLogin,name:" + deLogin.name);
            Log.d(TAG,"deLogin,passwd:" + deLogin.passwd);
            Log.d(TAG,"deLogin,userid:" + deLogin.userid);
        } catch (IOException e) {
            e.printStackTrace();
        }


    }
}

File: TestGlide/app/src/main/java/com/pix/testglide/MainActivity.kt
package com.pix.testglide

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Base64
import android.util.Log
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_main.*
import java.io.File
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity() {
    val imgUrl = "http://pixtang.com:8001/img/201812061333063264.png"
    val localUrl = "/storage/emulated/0/guagua/QiQi/chat/751046800dd149e4a6df96adcebd7a46.jpeg"
    var TAG = MainActivity::class.java.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        Glide.with(iv_img.context).load(imgUrl).into(iv_img)
        thread {
            var base64 = ImageUtilsKt.fileToBase64String(File(localUrl))
            Log.d(TAG, "base64:$base64")
            var imageByteArray = Base64.decode(base64, Base64.DEFAULT)
            iv_img.ktxRunOnUi { Glide.with(this@MainActivity).load(imageByteArray).asBitmap().into(this) }
        }
//        Glide.with(this).load(Uri.parse("file://$localUrl")).into(iv_img)
    }
}